src/nix/diff-closures.cc

```diff
@@ -25,15 +25,17 @@ GroupedPaths getClosureInfo(ref<Store> store, const StorePath & toplevel)
 
     GroupedPaths groupedPaths;
 
-    for (auto & path : closure) {
+    for (auto const & path : closure) {
         /* Strip the output name. Unfortunately this is ambiguous (we
            can't distinguish between output names like "bin" and
            version suffixes like "unstable"). */
         static std::regex regex("(.*)-([a-z]+|lib32|lib64)");
-        std::smatch match;
-        std::string name(path.name());
+        std::cmatch match;
```

Comment:
`smatch` might have been nice, but I don't see a way to make that work either. The std::regex does not seem to support it when the input is not an actual `std::string`. Unfortunate.